tasks/setup/InfracostSetup/index.ts

```diff
@@ -0,0 +1,284 @@
+  let path = `releases/download/v${version}`;
+  if (version === 'latest') {
+    path = `releases/latest/download`;
+  }
+
+  const platform = os.platform();
+  const filename = `infracost-${mapOS(platform)}-${mapArch(os.arch())}`;
+  const binaryName = platform === 'win32' ? 'infracost.exe' : filename;
+  const url = `https://github.com/infracost/infracost/${path}/${filename}.tar.gz`;
+
+  return {
+    url,
+    binaryName,
+  };
+}
+
+/**
+ * addBinaryToPath adds the cli into the pipeline tool path. This enables it for use in subsequent steps.
+ * If the binary is windows specific, it will rename infracost-<platform>-<arch> to infracost.
+ *
+ * @param pathToCLI - the full path to the cli executable
+ * @param binaryName - the binary name to use in the subsequent pipeline steps
+ */
+function addBinaryToPath(pathToCLI: string, binaryName: string) {
+  if (!binaryName.endsWith('.exe')) {
+    const source = path.join(pathToCLI, binaryName);
+    const target = path.join(pathToCLI, 'infracost');
+    taskLib.debug(`Moving ${source} to ${target}.`)
+
+    try {
+      taskLib.mv(source, target)
+    } catch (e) {
+      taskLib.error(`Unable to move ${source} to ${target}.`);
+      throw e;
+    }
+  }
+
+  toolLib.prependPath(pathToCLI);
+}
+
+/**
+ * getVersion returns a valid version from the task input version. If the input version is invalid it will attempt
+ * to find the closest valid version to the input version.
+ */
+async function getVersion(): Promise<string> {
+  const version = taskLib.getInput('version') ?? '';
+  if (semver.valid(version)) {
+    return semver.clean(version) || version;
+  }
+
+  if (!semver.validRange(version)) {
+    taskLib.warning(`${version} is not a valid version or range.`);
+    return version;
+  }
+
+  const max = semver.maxSatisfying(await getAllVersions(), version);
+  if (max) {
+    return semver.clean(max) || version;
+  }
+
+  taskLib.warning(`${version} did not match any release version.`);
+  return version;
+}
+
+/**
+ * ConfigOption is an interface defining an input option that the InfracostSetup task can take.
+ */
+interface ConfigOption {
+  inputName: string
+  configName: string
+  required: boolean
+  type: string
+}
+
+/**
+ * options defines a list of defined InfracostSetup task options. Defining how they correspond to cli configurations.
+ */
+const options: ConfigOption[] = [
+  {
+    inputName: 'apiKey',
+    configName: 'api_key',
+    required: true,
+    type: 'string'
+  },
+  {
+    inputName: 'currency',
+    configName: 'currency',
+    required: false,
+    type: 'string'
+  },
+  {
+    inputName: 'pricingApiEndpoint',
+    configName: 'pricing_api_endpoint',
+    required: false,
+    type: 'string'
+  },
+  {
+    inputName: 'enableDashboard',
+    configName: 'enable_dashboard',
+    required: false,
+    type: 'boolean'
+  }
+];
+
+async function configureInfracost() {
+  options.map(async (opt: ConfigOption) => {
```

Comment:
As this function doesn't return anything we probably can just iterate here
```suggestion
  options.forEach(async (opt: ConfigOption) => {
```

Reply:
💯